net.sourceforge.pmd.eclipse.plugin/src/main/java/net/sourceforge/pmd/eclipse/runtime/writer/impl/RuleSetWriterImpl.java

```diff
@@ -6,35 +6,38 @@
 
 import net.sourceforge.pmd.RuleSet;
 import net.sourceforge.pmd.RuleSetWriter;
+import net.sourceforge.pmd.RuleSets;
 import net.sourceforge.pmd.eclipse.runtime.writer.IRuleSetWriter;
 import net.sourceforge.pmd.eclipse.runtime.writer.WriterException;
 
 /**
- * Generate an XML rule set file from a rule set This class is a rewritting of
- * the original from PMD engine that doesn't support xpath properties !
+ * Generate an XML rule set file from a rule set This class is a rewritting of the original from PMD
+ * engine that doesn't support xpath properties !
  *
  * @author Philippe Herlin
  *
  */
 class RuleSetWriterImpl implements IRuleSetWriter {
 
-    /**
-     * Write a ruleset as an XML stream
-     * 
-     * @param writer
-     *            the output writer
-     * @param ruleSet
-     *            the ruleset to serialize
-     */
-    public void write(OutputStream outputStream, RuleSet ruleSet) throws WriterException {
-        try {
-            RuleSetWriter ruleSetWriter = new RuleSetWriter(outputStream);
-            ruleSetWriter.write(ruleSet);
-            outputStream.flush();
-        } catch (RuntimeException e) {
-            throw new WriterException(e);
-        } catch (IOException e) {
-            throw new WriterException(e);
-        }
+  /**
+   * Write a ruleset as an XML stream
+   * 
+   * @param writer
+   *          the output writer
+   * @param ruleSet
+   *          the ruleset to serialize
+   */
+  public void write(OutputStream outputStream, RuleSets ruleSets) throws WriterException {
+    try {
+      RuleSetWriter ruleSetWriter = new RuleSetWriter(outputStream);
+      for (RuleSet ruleSet : ruleSets.getAllRuleSets()) {
```

Comment:
The problem with writing multiple rulesets out is, that we probably don't have a reader, that can read this back...